src/fides/api/util/aws_util.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Optional
+
+from boto3 import Session
+from botocore.exceptions import ClientError
+from loguru import logger
+
+from fides.api.common_exceptions import StorageUploadError
+from fides.api.schemas.storage.storage import AWSAuthMethod, StorageSecrets
+
```

Comment:
Just a general comment on the utility here rather than any specific line of code: great idea to make a generic utility than can be reused, AWS auth won't change.

As I understand it, the utility gives three authn options:

1. Use an IAM user's long-term access key to authenticate for the entire session.
2. Rely on boto3's automatic method of checking the local environment for different ways AWS credentials can be provided by the env, and authenticate using them for the session. There are a few of these, ranging from env vars to using the EC2 instance metadata service, all documented by AWS [here].(https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html) 
3. Use an IAM user's long-term access key to get temporary security credentials (a.k.a. assume an IAM role) and use these temporary creds for the session.

Reply:
that's _just about_ spot on! the only slight difference is that i've also allowed the AWS credentials derived by boto3 (i.e. option #2) to be used in step #3, i.e. to assume an IAM role (if a user has specified `automatic` authentication method _while also_ providing an assume role ARN). 

is that even a valid use case? i was a bit unsure whether this is something that can (or should) be supported. if not, i can easily clean up the code and make this a bit more straightforward.